src/main/java/com/google/cloud/teleport/templates/common/SplunkConverters.java

```diff
@@ -183,6 +185,9 @@ public void processElement(ProcessContext context) {
                           String parsedTimestamp;
                           if (metadataAvailable) {
                             parsedTimestamp = metadata.optString(HEC_TIME_KEY);
+                          } else if (json.optJSONObject(DATA) != null
+                              && json.getJSONObject(DATA).has(TIMESTAMP_KEY)) {
+                            parsedTimestamp = json.getJSONObject(DATA).getString(TIMESTAMP_KEY);
```

Comment:
I think we want to make this condition restricted to `includePubsubMessage=true` since this fix is specific to that use case. Otherwise, this can lead to unexpected behavior where we're applying this logic to arbitrary payload that happens to have data.timestamp nested field (that is without PubsubMessage wrapper i.e. `includePubsubMessage=false`).

Reply:
That sounds like a good idea to me, too, @rarsan.

Reply:
This can be done but we'd be tying up `SplunkConverters` tightly with PubSub. 

The class is placed in a [common directory](https://github.com/GoogleCloudPlatform/DataflowTemplates/tree/master/src/main/java/com/google/cloud/teleport/templates/common) so other templates can make use of it in the future. Might be best to leave that parameter out if later on we decide to add a new template that has a different source with Splunk as the sink.

Reply:
@melbrodrigues - I see what you mean. Any ideas on loosening that coupling while keeping with the general spirit of the request?

Reply:
Discussed this with @rarsan yesterday and we feel it's better to add an additional check to see if the payload contains the `logName` field nested under `data`. (`logName` field is present for all Cloud Log events)

This should reduce chances of the template parsing the timestamp for arbitrary payloads.

Reply:
As discussed earlier, we won't be checking if `data.logName` field exists before parsing the timestamp to keep the behaviour consistent with `includePubsubMessage=false`.

Additionally, I've added a method `isPubsubMessage()` that checks for the `data` and `attributes` fields before parsing the timestamp.
```java
private boolean isPubsubMessage(JSONObject json) {
  return json.has(PUBSUB_MESSAGE_DATA_FIELD) && json.has(PUBSUB_MESSAGE_ATTRIBUTE_FIELD);
}
```

Reply:
Nice. LGTM.